recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+import os
+from os.path import join
+from glob import glob
+from itertools import chain
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class PclConanRecipe(ConanFile):
+    name = "pcl"
+    description = "Point Cloud Library"
+    license = "BSD-3-Clause"
+    homepage = "https://pointclouds.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cuda": [True, False],
+        "with_tools": [True, False]
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "with_cuda": False,
+        "with_tools": False
+    }
+    requires = ("boost/1.70.0",
+                "eigen/3.3.7",
+                "flann/1.9.1",
+                "libpng/1.6.37",
+                "opengl/system",
+                "qhull/7.3.2")
+    generators = ["cmake", "cmake_find_package"]
+    exports = ["CMakeLists.txt"]
+
+    @property
+    def _source_subfolder(self):
+        return "pcl-pcl-{}".format(self.version)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if tools.msvs_toolset(self) == "v140":
```

Comment:
```suggestion
        if self.options.shared:
            del self.options.fPIC
        if self.settings.compiler == "Visual Studio" and self.settings.compiler.version == "15":
```

Only to be sure, you want to restrict a compiler version or a toolset indeed?

Reply:
PCL won't compile with VS versions prior to 2017 (so toolset "v140" and previous). Hypothetically someone could be building with VS 2019 and the v140 toolset, so I check for the toolset.

Reply:
toolset versions prior to v140 should be handled by the VS version check in your suggested change.

Reply:
Indeed, sometimes I do it, to avoid multiple visual studio version running.

> PCL won't compile with VS versions prior to 2017 

So you need to check the compiler version too:

    if self.settings.compiler == "Visual Studio" and self.settings.compiler.version < "15":